SYCL/ESIMD/regression/copyto_char_test.cpp

```diff
@@ -0,0 +1,116 @@
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+//==- copyto_char_test.cpp - Test for using copy_to to copy char buffers -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+#include <iostream>
+#include <memory>
+
+using namespace sycl::ext::intel::esimd;
+
+template <typename DataT>
+using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
+template <typename DataT>
+using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
+
+using DataT = char;
+
+int main(int, char **) {
+  sycl::queue queue();
+
+  constexpr int NumElems = 10;
+  constexpr int NumSelectedElems = NumElems / 3;
+  constexpr int Stride = 2;
+  constexpr int Offset = 6;
+
+  shared_allocator<DataT> allocator(queue);
+  shared_vector<DataT> result(NumElems, allocator);
+  shared_vector<DataT> initial_ref_data(NumElems, allocator);
+  shared_vector<DataT> ref_data_for_fill(NumSelectedElems, allocator);
+
+  constexpr size_t value_for_increase_ref_data_for_fill = 50;
+  for (size_t i = 0; i < NumElems; ++i) {
+    initial_ref_data[i] = i + 1;
+  }
+  // We should have different values in the first reference data and in the
+  // second reference data.
+  for (size_t i = 0; i < NumSelectedElems; ++i) {
+    ref_data_for_fill[i] =
+        initial_ref_data[i] + value_for_increase_ref_data_for_fill;
+  }
+  queue.submit([&](sycl::handler &cgh) {
+    DataT *init_ref_ptr = initial_ref_data.data();
+    DataT *ref_data_for_fill_ptr = ref_data_for_fill.data();
+    DataT *const out_ptr = result.data();
+
+    cgh.single_task<class Kernel>([=]() SYCL_ESIMD_KERNEL {
+      simd<DataT, NumElems> src_simd_obj;
+      src_simd_obj.copy_from(init_ref_ptr);
+      simd<DataT, NumSelectedElems> dst_simd_obj;
+      dst_simd_obj.copy_from(ref_data_for_fill_ptr);
+      src_simd_obj.template select<NumSelectedElems, Stride>(Offset) =
+          dst_simd_obj;
+      src_simd_obj.copy_to(out_ptr);
+    });
+  });
+  queue.wait_and_throw();
+
+  std::vector<size_t> selected_indexsess;

```

Comment:
It looks like a misprint in the name of this var. It could be named 'selected_indices'